### ISSUE TITLE
Replace Alexa hotword with custom 'Sloppy' wake word

### ISSUE THREAD

**Issue by krystophny** [krystophny/tabura#649]
## Summary

Replace the vendored Alexa wake word model with a custom-trained "Sloppy" model using the existing OpenWakeWord training pipeline. The assistant name is "Sloppy" and the hotword should detect natural pronunciation of the word (short O, stress on first syllable — continental/British pronunciation, not American "slah-pee").

## Phasing

This issue covers **Stage 1** only. The full wake word journey has three stages:

### Stage 1: Automatic Piper-based pre-training (this issue)

- Fully automatic, no manual voice recording needed
- Uses existing Piper TTS multi-speaker pipeline to generate synthetic "sloppy" samples
- Trains an initial `sloppy.onnx` model via `scripts/train-hotword.sh`
- Wires the model into production (replaces `alexa.onnx`)
- Good enough for testing and development — detection will work but won't be tuned to the user's actual voice/accent
- **Goal**: get a working "sloppy" hotword deployed and testable ASAP

### Stage 2: Manual voice refinement (post Stage 1)

- Record 50-100 samples of the user saying "sloppy" in varied conditions
- Extract features and add to the training pipeline as custom positive data
- Retrain from the `train` step with the combined Piper + real voice data
- Iterate: test, record failure cases, retrain
- **This requires manual intervention** — someone needs to sit down and record samples

### Stage 3: Voice-cloning TTS augmentation via Tabura GUI (#651)

- The Tabura frontend provides a training GUI at `/hotword-train`
- User records voice samples directly in the browser
- State-of-the-art voice cloning models (Qwen3-TTS, GPT-SoVITS) are fine-tuned on the user's voice
- The cloned voice generates thousands of augmented "sloppy" samples automatically
- Combined with Piper diversity samples and openWakeWord augmentation pipeline
- Train, test in-browser, one-click deploy
- **Goal**: the voice-cloning AI takes over the heavy lifting after the user provides initial samples — no more manual iteration needed

Stage 1 is fully automatable and should be completed first. Stage 2 is a manual refinement step. Stage 3 (tracked in #651) makes Stage 2 unnecessary by automating voice-specific training through the Tabura UI.

## Motivation

- "Alexa" is a placeholder that collides with Amazon devices
- We want our own identity: "Sloppy"
- False positives are acceptable — the downstream STT + intent pipeline filters noise
- False negatives are not acceptable — the hotword must reliably trigger

### Target phrase

```yaml
model_name: "sloppy"

target_phrase:
  - "slop ee"

custom_negative_phrases:
  - "copy"
  - "happy"
  - "poppy"
  - "soapy"
  - "sleepy"
  - "slowly"
  - "sorry"
  - "soppy"
  - "stocky"
  - "rocky"
```

The phonetic representation `"slop ee"` produces a short rounded O as pronounced by German/Austrian/British speakers, not the American flat "slah-pee".

### Training parameters (favor recall over precision)

```yaml
n_samples: 5000
n_samples_val: 1000
augmentation_rounds: 5
layer_size: 64
steps: 100000
max_negative_weight: 1000
target_false_positives_per_hour: 1.0
```

Rationale: higher `target_false_positives_per_hour` (1.0 vs current 0.2) and more training samples/steps produce a model that errs toward detection. The STT + intent layer downstream handles false activations.

### Prerequisites and setup

1. **Python 3.12** with venv support
2. **CUDA GPU** recommended (CPU training works but is ~10x slower)
3. **~10 GB disk** for downloaded datasets (ACAV100M features, AudioSet, FMA, Piper model)

Run:
```bash
scripts/train-hotword.sh
```

This automatically:
- Clones `openwakeword-trainer` into `tools/openwakeword-trainer/` if missing
- Creates a Python venv with all dependencies (including `piper-phonemize-fix` for Python 3.12+)
- Runs the 13-step pipeline: env check, download data, generate clips via Piper TTS, augment, extract features, train DNN, export ONNX

### Adding custom voice recordings to refine the model

After the initial Piper-based model is trained and working, refine it with real voice recordings:

1. **Record samples** — 50-100 recordings of yourself saying "sloppy" in varied tones, speeds, distances from mic. Save as 16 kHz mono WAV files.

2. **Place recordings** in a directory, e.g. `data/custom_positive/`

3. **Create a feature extraction script** that runs the recordings through the mel-spectrogram and embedding models to produce `.npy` feature files matching the format of `openwakeword_features_ACAV100M_2000_hrs_16bit.npy`

4. **Add to config**:
   ```yaml
   feature_data_files:
     "ACAV100M_sample": "data/openwakeword_features_ACAV100M_2000_hrs_16bit.npy"
     "custom_positive": "data/custom_positive_features.npy"

   batch_n_per_class:
     "ACAV100M_sample": 1024
     "adversarial_negative": 50
     "positive": 50
     "custom_positive": 25
   ```

5. **Re-train** from the `train` step:
   ```bash
   scripts/train-hotword.sh --from-step train
   ```

6. **Iterate** — record failure cases (phrases that should trigger but don't), add them, retrain.

The custom recordings will anchor the model to the actual voice/accent while the Piper synthetic data provides breadth across many speakers.

### Update hotword-config.yaml

Replace `scripts/hotword-config.yaml` with the sloppy config above.

### Update model references

**`internal/web/hotword.go`**:
```go
const hotwordModelFileName = "sloppy.onnx"
```

**`internal/web/static/hotword.ts`**:
```typescript
const HOTWORD_MODEL_FILES = {
  mel: `${HOTWORD_VENDOR_BASE}/melspectrogram.onnx`,
  embedding: `${HOTWORD_VENDOR_BASE}/embedding_model.onnx`,
  keyword: `${HOTWORD_VENDOR_BASE}/sloppy.onnx`,
};
```

### Lower default threshold

**`internal/web/static/hotword.ts`**:
```typescript
const HOTWORD_DEFAULT_THRESHOLD = 0.3;
```

Rationale: favors recall (no false negatives) over precision (some false positives OK).

### Reduce detection cooldown

```typescript
const HOTWORD_DETECTION_COOLDOWN_MS = 800;
```

Shorter cooldown allows faster re-trigger after a detection that the STT layer discards.

### Deploy model files

Copy trained model to vendored location:
```bash
cp models/hotword/sloppy.onnx internal/web/static/vendor/openwakeword/sloppy.onnx
```

Remove old model:
```bash
rm internal/web/static/vendor/openwakeword/alexa.onnx
```

The mel-spectrogram and embedding models (`melspectrogram.onnx`, `embedding_model.onnx`) are shared across all keywords and stay as-is.

### Update status endpoint

`internal/web/hotword.go` — `hotwordRuntimeAssetFiles` slice already derives from `hotwordModelFileName`, so changing the const is sufficient.

### Training progress (Stage 1 — CLI)

Stage 1 training runs via `scripts/train-hotword.sh` on the command line. The 13-step pipeline prints progress to stdout:

1. Steps 1-5 (env, patches, download, verify, config) — fast, ~1 minute total
2. Step 6 (generate clips via Piper TTS) — ~10 minutes, prints per-batch progress
3. Steps 7-8 (resample, verify clips) — instant
4. Step 9 (augment + extract features) — ~30 minutes, prints per-round progress
5. Steps 10 (verify features) — instant
6. Step 11 (train DNN) — ~30 minutes, prints loss every 1000 steps
7. Steps 12-13 (verify model, export) — instant

Total: ~1-1.5 hours with GPU, ~10 hours CPU-only.

The script exits 0 on success with `trained model: models/hotword/sloppy.onnx`. On failure, it prints the failed step name and exit code, and can be resumed with `--from-step <step>`.

### Deployment and hot-reload

After training completes:

1. Copy `sloppy.onnx` to `internal/web/static/vendor/openwakeword/`
2. The browser can hot-reload the new model without page refresh — `initHotword({ force: true })` tears down existing ONNX sessions and re-loads from the updated files
3. Add a backend endpoint `POST /api/hotword/reload` that:
   - Verifies the new model file exists and is valid
   - Sends a WebSocket message to all connected clients to trigger `initHotword({ force: true })`
   - Returns the new model status (file size, timestamp)
4. The status indicator should briefly show "Hotword model updated" when the reload completes

### Status endpoint

`GET /api/hotword/status` already reports model availability. Extend it to include:
- Model file name and size
- Model file modification timestamp (so the user can tell whether they're running the old or new model)
- Whether a training run is in progress (for future GUI integration)

### Unit tests

- `internal/web/hotword_test.go` — update expected model filename from `alexa.onnx` to `sloppy.onnx`

### Playwright tests

- `tests/playwright/hotword.spec.ts` — all 10 existing tests should pass with the mock injection path unchanged (mock doesn't depend on model name)
- Add a test that verifies the keyword model URL contains `sloppy.onnx`

### Manual testing

- Use `internal/web/static/hotword-test.html` with the new model
- Test with threshold slider at 0.3
- Verify detection at various distances, volumes, and speaking speeds
- Verify "sloppy" triggers standalone, at beginning of sentence, and mid-sentence

### Acceptance criteria

- [ ] `sloppy.onnx` trained and vendored
- [ ] All references to `alexa.onnx` removed
- [ ] Default threshold lowered to 0.3
- [ ] Cooldown reduced to 800ms
- [ ] `hotword-config.yaml` updated for sloppy
- [ ] All Go tests pass
- [ ] All Playwright hotword tests pass
- [ ] Manual testing confirms detection with continental pronunciation
- [ ] `scripts/train-hotword.sh` produces `sloppy.onnx` end-to-end
